src/Vixen.Common/ffmpeg/ffmpeg.cs

```diff
@@ -1,92 +1,99 @@
 ﻿using System.Diagnostics;
 using System.Globalization;
+using System.Text.RegularExpressions;
 
 namespace Vixen.Common.ffmpeg
 {
-	public class Ffmpeg
+	public partial class Ffmpeg
 	{
 		private static readonly string FfmpegPath;
 
+		//ffmpeg output line: "  Duration: 00:02:29.46, start: 0.00...."
+		[GeneratedRegex(@"Duration: (\d+):(\d{2}):(\d{2})\.(\d{2})")]
+		private static partial Regex parseDuration();
+
+		// look for ", ####x####" where numbers can be 2 to 5 digits long
+		[GeneratedRegex(@", (?<width>\d{2,5})x(?<height>\d{2,5})")]
+		private static partial Regex parseResolution();
+
 		static Ffmpeg()
 		{
 			FfmpegPath = Path.Combine(AppDomain.CurrentDomain.BaseDirectory, @"ffmpeg.exe");
 		}
 
-		//Nutcracker Video Effect
-		public static void MakeThumbnails(string movieFile, int width, int height, string outputPath, int framesPerSecond = 20)

```

Comment:
Removing public methods is generally not a good idea since you may not know exactly what other code may be using it. Keep in mind that there may be plugin code outside of Vixen that we do not control and this could introduce breaking changes. Recommend leaving the old signature and try to massage it to wrap and call the new implementation, or leave the old code and mark it deprecated with info on using the replacement.

Reply:
Ahh, that makes sense - I did search the Vixen code base for any usage and found none, but didn't realize there can be external plugin code that still might use these functions. I'll restored the original functions, but marked them as Obsolete.